src/main/java/net/dv8tion/jda/api/entities/MessageReaction.java

```diff
@@ -203,21 +205,20 @@ public PrivateChannel getPrivateChannel()
      * @return The channel this Reaction was used in
      */
     @Nonnull
-    public MessageChannel getChannel()
+    public MessageChannelUnion getChannel()
     {
-        return channel;
+        return (MessageChannelUnion) channel;
```

Comment:
Should `channel` not be this type instead of just casting every time its used?

Reply:
So changing this to store the channel as the union instead of the supertype has 2 effects:
1) We have to update all the code passing channels _in_ to do the casts. This ends up touching a _lot_ more places than casting at getter site
2) For some situations, like in MessageReaction, storing it as `MessageChannelUnion` would still result in a cast because `getGuildChannel` needs a cast to `GuildMessageChannelUnion`.

Where possible I've taken the mentality of accepting the "base" interface (`MessageChannel` in this case) and handling the messiness of union upcasting in the getters as it seems like the easiest way to prevent passing unions around everywhere and forcing casts everywhere.

Reply:
Couldn't most of the casts be done in the constructor though?

Reply:
I've elected to go forward with return site casting so that unions don't need to be distributed throughout the system.